### PR TITLE
kill: Don't unlock nodes if killing procs fails

### PR DESCRIPTION
**Pull request by zmc**
... so that we don't unlock nodes while their jobs are running.

### CODE REVIEW

File: teuthology/kill.py [hunk 1]
```diff
@@ -187,6 +189,7 @@ def kill_processes(run_name, pids=None):
     if len(to_kill) == 0:
         log.info("No teuthology processes running")
     else:
+        survivors = list()
```
Comment (kamoltat) [inline teuthology/kill.py:192]: Can I ask why `list()` instead of `[]`. I don't mind `list()`, just curious.

**Review by kamoltat — APPROVED** [review]
LGTM, just one minor comment that I'm just curious about.

File: teuthology/kill.py (hunk 1 above)
Comment (zmc) [reply teuthology/kill.py:192]: Good question! It wasn't actually intentional, and `[]` is faster, so I'll replace it :)

**Review by kamoltat — APPROVED** [review]
Looks good!

**Pull request merged**